test/mpi/coll/nonblocking2.c

```diff
@@ -64,16 +66,35 @@ int main(int argc, char **argv)
     rdispls = malloc(size * sizeof(int));
     sendtypes = malloc(size * sizeof(MPI_Datatype));
     recvtypes = malloc(size * sizeof(MPI_Datatype));
+    test_icoll_with_root(rank, size, 0);
+    test_icoll_with_root(rank, size, size - 1);
+
+    MTest_Finalize(0);
+    free(buf);
+    free(recvbuf);
+    free(sendcounts);
+    free(recvcounts);
+    free(rdispls);
+    free(sdispls);
+    free(recvtypes);
+    free(sendtypes);
+    return 0;
+}
+
+void test_icoll_with_root(int rank, int size, int root)
+{
+    int i, j;
 
+    root = size - 1;
```

Comment:
overriding input arg?

Reply:
Thanks for pointing out my negligence. I'll update it.

Reply:
Updated.